tuist/tuist
# Add support for `systemLibrary` SwiftPackageManager targets

Resolves https://github.com/tuist/tuist/issues/3477

### Short description 📝
This pull request attempts to resolve the issue with system library targets not being correctly linked.

`PackageInfoMapper` implementation was updated to process `.system` targets and search for corresponding module map file similar to SPM implementation.

### How to test the changes locally 🧐
- download a sample project which includes a basic setup for targets depending on GRDB (https://drive.google.com/file/d/1FiK0NMeOlI63-6cX16A1ZwR5HkYrxcp3/view?usp=sharing)
- run `tuist fetch` and `tuist generate`
- observe the project is generated with no errors and can be successfully run. Current tuist version would display a `GRDB is not a valid configured external dependency` error.

### Checklist ✅
